### PR TITLE
fix: avoid panic in user audit event resolver on nil event data

### PR DESCRIPTION
**Pull request by Aloui-Ikram**
The user audit-event resolver is registered for both the /api/v2.0/users collection and the /api/v2.0/users/{id} resource. For a PUT whose URL does not match a specific user id, the base resolver returns without populating event.Data, leaving it nil. userEventResolver.Resolve then dereferenced event.Data through an unchecked type assertion, which panics.

Audit events are built and published in a detached goroutine (BuildAndPublish) that had no recover(), so the panic propagated and terminated the process instead of failing the single request.

- user.go: use a checked type assertion and skip the update when event.Data is nil.
- event.go: recover() in the BuildAndPublish goroutine as a safeguard.

Add a regression test covering the nil event.Data path.

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/pkg/notifier/event/event.go [hunk 1]
```diff
@@ -100,6 +100,13 @@ func (e *Event) Publish(ctx context.Context) error {
 // The process is done in a separated goroutine
 func BuildAndPublish(ctx context.Context, metadata ...Metadata) {
 	go func() {
+		// recover in case of panic during building or publishing the event,
+		// otherwise a panic in this detached goroutine would crash the process
+		defer func() {
+			if err := recover(); err != nil {
+				log.Errorf("recovered from the panic when building and publishing event: %v", err)
+			}
```
[the remark this answers is not in the document]
Comment (bupd) [reply src/pkg/notifier/event/event.go]: i would not recommend the `debug.Stack()` but to keep things consistent use the `runtime.Stack()` used around  codebase. 

@Aloui-Ikram

[the remark this answers is not in the document]
Comment (Aloui-Ikram) [reply src/pkg/notifier/event/event.go]: Done, switched to runtime.Stack() 
thanks @bupd

**Review by Vad1mo — APPROVED** [review]

**Review by bupd — APPROVED** [review]
/lgtm
